src/bin/pgcopydb/dump_restore.c

```diff
@@ -567,6 +570,167 @@ copydb_target_finalize_schema(CopyDataSpec *specs)
 }
 
 
+typedef struct ArchiveContext
+{
+	CopyDataSpec *specs;
+	PQExpBuffer line;
+	FILE *listFile;
+} ArchiveContext;
+
+static bool
+copydb_write_restore_list_callback(void *ctx, ArchiveContentItem *item)
```

Comment:
Missing function-level comments. Also, consider using `_hook` suffix instead of `_callback` like every other place in the rest of the code. Same concept, same name, helps maintenance (cognitive overhead).